pkgs/applications/networking/cluster/terraform/default.nix

```diff
@@ -28,45 +29,58 @@ let
 
       ldflags = [ "-s" "-w" ];
 
+      subPackages = [ "." ];
+
       postConfigure = ''
         # speakeasy hardcodes /bin/stty https://github.com/bgentry/speakeasy/issues/22
         substituteInPlace vendor/github.com/bgentry/speakeasy/speakeasy_unix.go \
           --replace "/bin/stty" "${coreutils}/bin/stty"
       '';
 
+      nativeBuildInputs = [ installShellFiles ];
+
+      preCheck = ''
+        export HOME=$TMPDIR
+        export TF_SKIP_REMOTE_TESTS=1
+      '';
+
+      # https://github.com/posener/complete/blob/9a4745ac49b29530e07dc2581745a218b646b7a3/cmd/install/bash.go#L8
+      completionBash = "complete -C terraform terraform\n";
+      # https://github.com/posener/complete/blob/9a4745ac49b29530e07dc2581745a218b646b7a3/cmd/install/zsh.go
+      completionZsh =  "complete -C terraform terraform\n";
+      # https://github.com/posener/complete/blob/9a4745ac49b29530e07dc2581745a218b646b7a3/cmd/install/fish.go#L56
+      completionFish = ''
+        function __complete_terraform
+            set -lx COMP_LINE (commandline -cp)
+            test -z (commandline -ct)
+            and set COMP_LINE "$COMP_LINE "
+            terraform
+        end
+        complete -f -c terraform -a "(__complete_terraform)"
+      '';
+
+      passAsFile = [ "completionBash" "completionZsh" "completionFish" ];
+
       postInstall = ''
         # remove all plugins, they are part of the main binary now
         for i in $out/bin/*; do
           if [[ $(basename $i) != terraform ]]; then
             rm "$i"
           fi
         done
-      '';
 
-      preCheck = ''
-        export HOME=$TMPDIR
-        export TF_SKIP_REMOTE_TESTS=1
+        installShellCompletion --bash --name terraform $completionBashPath
+        installShellCompletion --zsh --name terraform $completionZshPath
+        installShellCompletion --fish --name terraform $completionFishPath
```

Comment:
It does not work because you mixed the vars up here.

Reply:
Can you be more specific about what you think is incorrect here please?